src/rules/package-size.js

```diff
@@ -47,52 +47,52 @@ class PackageSize extends Adviser.Rule {
 
   async run(sandbox) {
     const packagejson = require(path.join(this.context.filesystem.dirname, 'package.json'));
-    const packages = [];
+    const processing = { packages: [], stats: [] };
     const promises = [];
-    const results = [];
-    const skip = [];
+    const results = { complete: [], skip: [] };
 
     if (packagejson.hasOwnProperty('dependencies')) {
       const dependencies = Object.keys(packagejson['dependencies']);
-      const versionFilter = new RegExp('[^0-9.]+');
 
-      try {
-        // attempting to identify the most accurate version number possible using package-lock.json
-        const packagelockjson = require(path.join(this.context.filesystem.dirname, 'package-lock.json'));
-
-        dependencies.forEach((name, index) => {
-          const version = packagelockjson['dependencies'][name]['version'].replace(versionFilter, '');
-          packages.push({ name: name, version: version });
-        });
-      } catch {
-        dependencies.forEach((name, index) => {
-          const version = packagejson['dependencies'][name].replace(versionFilter, '');
-          packages.push({ name: name, version: version });
-        });
-      }
+      dependencies.forEach((name, index) => {
+        if (!this.parsedOptions.whitelist.includes(name)) {
+          const dependency = { name: name, version: '' };
+
+          try {
+            dependency.version = require(`${this.context.filesystem.dirname}/node_modules/${
```

Comment:
Here is better to use `resolve` to get the module path, and then require the `package.json` inside. To concatenate paths please use the module `path`, to avoid issues in other OSs

Reply:
@iranreyes As per our discussion, I have addressed this issue using `path.join()` as opposed to `resolve`. The reason is that `resolve` returns the package entry point, which is not always in the same directory as the `package.json` file.